src/UIKit/UIApplication.cs

```diff
@@ -50,11 +50,21 @@ partial class UIApplication
 		// We link with __Internal here so that this function is interposable from third-party native libraries.
 		// See: https://github.com/xamarin/MicrosoftInTune/issues/3 for an example.
 		[DllImport ("__Internal")]
+#if NET
+		extern static int xamarin_UIApplicationMain (int argc, /* char[]* */ IntPtr argv, /* NSString* */ IntPtr principalClassName, /* NSString* */ IntPtr delegateClassName, out IntPtr gchandle);
```

Comment:
Is there any reason we can't always use the new code (i.e. don't check for NET)?